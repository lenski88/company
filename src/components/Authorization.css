.Authorization {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  border: 0.2em solid  #fbfc55;
  padding: 3em;
  margin-top:25vh
}

.Authorization .inputText {
    border-color: #fbfc55;
    background-color: #fbfc55 ;
    color:#0301a8;
    font-family:  'Press Start 2P', cursive;
    padding: 0.3em;
}
