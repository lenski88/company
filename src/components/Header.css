
.Control {
    box-sizing: border-box;
    position: absolute;
    width: 100%;
    border: 0.1em solid;
    color: #fbfc55;
    background-color: #0301a8;
}

 .Control .inputText {
    border-color: #fbfc55;
    background-color: #fbfc55 ;
    color:#0301a8;
    font-family:  'Press Start 2P', cursive;
    padding: 0.3em;
}

.NoControl {
    display: none
}