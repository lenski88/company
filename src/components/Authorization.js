import React from "react";
import PropTypes from "prop-types";

import "./Authorization.css";

class Authorization extends React.PureComponent {
  static propTypes = {
    employes: PropTypes.array.isRequired /* массив сотрудников */,
    login:
      PropTypes.bool.isRequired /* прошла ли авторизация; по умолчанию false */,
    cbGetUser: PropTypes.func,
  };

  state = {
    login: this.props.login,
    nameUser: "",
    errName: "",
    button: false,
  };

  nameUser = (eo) => {
    this.setState({
      nameUser: eo.target.value,
    });

    let arrayNames = this.props.employes;
    let name = eo.target.value;
    let isLogin = arrayNames.some((i) => {
      return name === i.login;
    });

    if (!isLogin) {
      this.setState({
        errName: "Нет такого сотрудника",
      });
    } else {
      this.setState({
        errName: "",
      });
    }
  };

  getUser = () => {
    if (!this.state.errName && !this.state.errPassword) {
      let arrayNames = this.props.employes;
      let name = this.state.nameUser;
      let user = arrayNames.find((i) => {
        return name === i.login;
      });
      this.props.cbGetUser(user);
      this.setState({
        nameUser: "",
        login: this.props.login,
      });
    }
  };

  render() {
    return (
      !this.state.login && (
        <div className={!this.props.login?"Authorization": "NoAuthorization"}>
          <label>Введите логин:</label>
          <br />
          <input
            className="inputText"
            type="text"
            name="nameUser"
            value={this.state.nameUser}
            onChange={this.nameUser}
            autoFocus
          ></input>
          <br />
          <span>{this.state.errName}</span>
          <br />
          <br />
          {this.state.nameUser && (
            <input
              className="button"
              type="button"
              onPointerDown={this.getUser}
              value="[Войти]"
              disabled={this.state.button}
            ></input>
          )}
        </div>
      )
    );
  }
}

export default Authorization;
