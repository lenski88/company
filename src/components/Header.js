import React from "react";
import PropTypes from "prop-types";

import AddNewEmp from "./AddNewEmp";

import "./Header.css";

class Header extends React.PureComponent {
  static propTypes = {
    employes: PropTypes.array.isRequired,
    login: PropTypes.bool.isRequired,
    user: PropTypes.object,
    cbExit: PropTypes.func,
    cbSearchDepartment: PropTypes.func,
    cbNewEmpPush:PropTypes.func
  };

  state = {
    addEmp: 0, // 0 - по умолчанию, 1 - добавление нового сотрудника
  };

  exit = () => {
    this.props.cbExit({
      login: false,
    });
    this.setState({
      addEmp: 0,
    });
  };

  selectDep = (eo) => {
    this.props.cbSearchDepartment({ select: eo.target.value });
  };

  addEmp = () => {
    this.setState({
      addEmp: 1,
    });
  };

  exitAdd = () => {
    this.setState({
      addEmp: 0,
    });
  };

  newEmp = (newEmp) => {
    this.props.cbNewEmpPush(newEmp)
  }

  render() {
    let name; // имя юзера
    let countTask; // количество задач на кнопке; равно длине массива задач
    if (this.props.user) {
      name = this.props.user.name;
      name = name.split(" ");
      name = name[1];

      countTask = `Мои задачи [${this.props.user.task.length}]`;
    }
    return (
      this.props.login && (
        <div className="Control">
          <div style={{ display: "flex", justifyContent: "center" }}>
            <h3>Приветствую, {name}!</h3>
          </div>
          <div style={{ display: "flex", justifyContent: "space-around" }}>
            {!this.state.addEmp && (
              <div>
                <span>Искать по отделам:</span>
                <br />
                <br />
                <select className="inputText" onChange={this.selectDep}>
                  <option value="Все">Все</option>
                  <option value="Руководство">Руководство</option>
                  <option value="Отдел закупок">Отдел закупок</option>
                  <option value="Отдел продаж">Отдел продаж</option>
                  <option value="Транспортный отдел">Транспортный отдел</option>
                  <option value="Бухгалтерия">Бухгалтерия</option>
                  <option value="Отдел кадров">Отдел кадров</option>
                  <option value="Складское хозяйство">
                    Складское хозяйство
                  </option>
                </select>
              </div>
            )}
            {this.props.user.level === 3 && !this.state.addEmp && (
              <input
                className="button"
                type="button"
                value="Добавить сотрудника"
                onPointerDown={this.addEmp}
              ></input>
            )}
            {!this.state.addEmp && (
              <input className="button" type="button" value={countTask}></input>
            )}
            {!this.state.addEmp && (
              <input
                className="button"
                type="button"
                value="Выйти"
                onPointerDown={this.exit}
              ></input>
            )}
          </div>
          {Boolean(this.state.addEmp) && (
            <AddNewEmp employes={this.props.employes} cbExit={this.exitAdd} cbAddEmp={this.newEmp} />
          )}
        </div>
      )
    );
  }
}

export default Header;
