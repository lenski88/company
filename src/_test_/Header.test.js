"use strict";

import React from "react";
import renderer from "react-test-renderer";
import { employes } from "./Authorization.test";
import { user } from "./Authorization.test";

import Header from "../components/Header";

test("проверка header", () => {
  let component = renderer.create(
    <Header user={user} employes={employes} login={true} modeEmployes={0} />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

  let newUser = {
    id: 58,
    login: "kuzmin58",
    status: 0,
    level: 1,
    name: "Кузьмин Цицерон Данилович",
    email: "kuzminC",
    phone: "339043722",
    department: "Складское хозяйство",
    position: "Грузчик",
    task: [],
  };

  component = renderer.create(
    <Header user={newUser} employes={employes} login={true} modeEmployes={0} />
  );

  componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

  component = renderer.create(
    <Header user={newUser} employes={employes} login={true} modeEmployes={1} />
  );

  componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

  component = renderer.create(
    <Header user={newUser} employes={employes} login={false} modeEmployes={0} />
  );

  componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test("правильность отображения имени", () => {
  let component = renderer.create(
    <Header user={user} employes={employes} login={true} modeEmployes={0} />
  );

  let helloUser = component.root.find((el) => el.props.id == "helloUser");
  expect(helloUser.children.join("")).toBe("Приветствую, Мишель!");

  let newUser = {
    id: 40,
    login: "kulakovI40",
    status: 0,
    level: 2,
    name: "Кулаков Иван Иванович",
    email: "kulakov@example.com",
    phone: "294389052",
    department: "Складское хозяйство",
    position: "Начальник отдела",
    task: [],
  };

  component = renderer.create(
    <Header user={newUser} employes={employes} login={true} modeEmployes={0} />
  );

  helloUser = component.root.find((el) => el.props.id == "helloUser");
  expect(helloUser.children.join("")).toBe("Приветствую, Иван!");
});

test("правильность отображения количества задач на кнопке", () => {
  let user = {
    id: 0,
    login: "ivanov0",
    status: 0,
    level: 3,
    name: "Иванов Иван Петрович",
    email: "ivanov@example.com",
    phone: "293345422",
    department: "Руководство",
    position: "Директор",
    task: [
      {
        id: 1,
        task: "Сделайте что-нибудь",
        sender: "Кулаков Кондрат Иванович",
      },
      {
        id: 2,
        task: "Поднимите уже, наконец, зарплату!",
        sender: "Ковалёва Радмила Даниловна",
      },
      { id: 3, task: "Погладьте котика", sender: "Шаров Ян Александрович" },
      { id: 4, task: "Погладьте котика", sender: "Шаров Ян Александрович" },
      { id: 5, task: "Погладьте котика", sender: "Шаров Ян Александрович" },
      { id: 6, task: "Погладьте котика", sender: "Шаров Ян Александрович" },
      { id: 7, task: "Погладьте котика", sender: "Шаров Ян Александрович" },
    ],
  };

  let component = renderer.create(
    <Header user={user} employes={employes} login={true} modeEmployes={0} />
  );
  let buttonTask = component.root.find((el) => el.props.id == "buttonTask");
  expect(buttonTask.props.value).toBe("[Мои задачи]:[7]");


  let newUser = {
    id: 0,
    login: "ivanov0",
    status: 0,
    level: 3,
    name: "Иванов Иван Петрович",
    email: "ivanov@example.com",
    phone: "293345422",
    department: "Руководство",
    position: "Директор",
    task: [
      {
        id: 1,
        task: "Сделайте что-нибудь",
        sender: "Кулаков Кондрат Иванович",
      },
      {
        id: 2,
        task: "Поднимите уже, наконец, зарплату!",
        sender: "Ковалёва Радмила Даниловна",
      },
      { id: 3, task: "Погладьте котика", sender: "Шаров Ян Александрович" },
    ],
  };

  component = renderer.create(
    <Header user={newUser} employes={employes} login={true} modeEmployes={0} />
  );
   buttonTask = component.root.find((el) => el.props.id == "buttonTask");
  expect(buttonTask.props.value).toBe("[Мои задачи]:[3]");
});
