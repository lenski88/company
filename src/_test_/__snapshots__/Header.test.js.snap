// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка Authorization 1`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  
</div>
`;

exports[`проверка Authorization 2`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="sdfasdf"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка Authorization 3`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка Authorization 4`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="saveliev47"
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка header 1`] = `
<div
  className="Control"
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <h3
      id="helloUser"
    >
      Приветствую, 
      Мишель
      !
    </h3>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <div>
      <span>
        Искать по отделам:
      </span>
      <br />
      <br />
      <select
        className="inputText"
        onChange={[Function]}
      >
        <option
          value="Все"
        >
          Все
        </option>
        <option
          value="Руководство"
        >
          Руководство
        </option>
        <option
          value="Отдел закупок"
        >
          Отдел закупок
        </option>
        <option
          value="Отдел продаж"
        >
          Отдел продаж
        </option>
        <option
          value="Транспортный отдел"
        >
          Транспортный отдел
        </option>
        <option
          value="Бухгалтерия"
        >
          Бухгалтерия
        </option>
        <option
          value="Отдел кадров"
        >
          Отдел кадров
        </option>
        <option
          value="Складское хозяйство"
        >
          Складское хозяйство
        </option>
      </select>
    </div>
    <input
      className="button"
      onPointerDown={[Function]}
      type="button"
      value="[Добавить сотрудника]"
    />
    <input
      className="button"
      id="buttonTask"
      onPointerDown={[Function]}
      type="button"
      value="[Мои задачи]:[0]"
    />
    <input
      className="button"
      onPointerDown={[Function]}
      type="button"
      value="[Выйти]"
    />
  </div>
</div>
`;

exports[`проверка header 2`] = `
<div
  className="Control"
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <h3
      id="helloUser"
    >
      Приветствую, 
      Цицерон
      !
    </h3>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <div>
      <span>
        Искать по отделам:
      </span>
      <br />
      <br />
      <select
        className="inputText"
        onChange={[Function]}
      >
        <option
          value="Все"
        >
          Все
        </option>
        <option
          value="Руководство"
        >
          Руководство
        </option>
        <option
          value="Отдел закупок"
        >
          Отдел закупок
        </option>
        <option
          value="Отдел продаж"
        >
          Отдел продаж
        </option>
        <option
          value="Транспортный отдел"
        >
          Транспортный отдел
        </option>
        <option
          value="Бухгалтерия"
        >
          Бухгалтерия
        </option>
        <option
          value="Отдел кадров"
        >
          Отдел кадров
        </option>
        <option
          value="Складское хозяйство"
        >
          Складское хозяйство
        </option>
      </select>
    </div>
    <input
      className="button"
      id="buttonTask"
      onPointerDown={[Function]}
      type="button"
      value="[Мои задачи]:[0]"
    />
    <input
      className="button"
      onPointerDown={[Function]}
      type="button"
      value="[Выйти]"
    />
  </div>
</div>
`;

exports[`проверка header 3`] = `
<div
  className="NoControl"
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <h3
      id="helloUser"
    >
      Приветствую, 
      Цицерон
      !
    </h3>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <div>
      <span>
        Искать по отделам:
      </span>
      <br />
      <br />
      <select
        className="inputText"
        onChange={[Function]}
      >
        <option
          value="Все"
        >
          Все
        </option>
        <option
          value="Руководство"
        >
          Руководство
        </option>
        <option
          value="Отдел закупок"
        >
          Отдел закупок
        </option>
        <option
          value="Отдел продаж"
        >
          Отдел продаж
        </option>
        <option
          value="Транспортный отдел"
        >
          Транспортный отдел
        </option>
        <option
          value="Бухгалтерия"
        >
          Бухгалтерия
        </option>
        <option
          value="Отдел кадров"
        >
          Отдел кадров
        </option>
        <option
          value="Складское хозяйство"
        >
          Складское хозяйство
        </option>
      </select>
    </div>
    <input
      className="button"
      id="buttonTask"
      onPointerDown={[Function]}
      type="button"
      value="[Мои задачи]:[0]"
    />
    <input
      className="button"
      onPointerDown={[Function]}
      type="button"
      value="[Выйти]"
    />
  </div>
</div>
`;

exports[`проверка header 4`] = `null`;

exports[`проверка нажатия на кнопку войти 1`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  
</div>
`;

exports[`проверка нажатия на кнопку войти 2`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка нажатия на кнопку войти 3`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка нажатия на кнопку войти 4`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка рендера кнопки при верном логине 1`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  
</div>
`;

exports[`проверка рендера кнопки при верном логине 2`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;
